src/main/java/org/apache/commons/collections4/map/CaseInsensitiveMap.java

```diff
@@ -141,6 +141,70 @@ protected Object convertKey(final Object key) {
         return AbstractHashedMap.NULL;
     }
 
+    /**
+     * Puts a key-value mapping into this map.
+     *
+     * @param key  the key to add
+     * @param value  the value to add
+     * @return the value previously mapped to this key, null if none
+     */
+    @Override
+    public V put(final K key, final V value) {
+        final Object convertedKey = convertKey(key);
+        final int hashCode = hash(convertedKey);
+        final int index = hashIndex(hashCode, data.length);
+        HashEntry<K, V> entry = data[index];
+        while (entry != null) {
+            if (entry.hashCode == hashCode && isEqualKey(convertedKey, entry.key)) {
+                final V oldValue = entry.getValue();
+                updateEntry(entry, value);
+                return oldValue;
+            }
+            entry = entry.next;
+        }
+
+        addMappingWithoutKeyConversion(index, hashCode, convertedKey, value);
+        return null;
+    }
+
+    /**
+     * Adds a new key-value mapping into this map.
+     * <p>
+     * This implementation calls {@code createEntry()}, {@code addEntry()}
```

Comment:
This javadoc has been copied from AbstractHashedMap but is not correct since the method does not call createEntry.

It has a reference to sub-classes overriding to fully control adds to the map. They cannot override createEntryWithoutKeyConversion or addMappingWithoutKeyConversion as they are private. They could override addEntry and checkCapacity to partially control adds to the map.